examples/data-objects/task-selection/src/containerCode.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { BaseContainerRuntimeFactory } from "@fluidframework/aqueduct";
+import { IContainerRuntime } from "@fluidframework/container-runtime-definitions";
+import { RuntimeRequestHandler } from "@fluidframework/request-handler";
+import { RequestParser, requestFluidObject } from "@fluidframework/runtime-utils";
+
+import { OldestClientDiceRoller, OldestClientDiceRollerInstantiationFactory } from "./oldestClientDiceRoller";
+import { TaskManagerDiceRoller, TaskManagerDiceRollerInstantiationFactory } from "./taskManagerDiceRoller";
+import { IDiceRoller } from "./interface";
+
+const registryEntries = new Map([
+    OldestClientDiceRollerInstantiationFactory.registryEntry,
+    TaskManagerDiceRollerInstantiationFactory.registryEntry,
+]);
+
+export const taskManagerDiceId = "taskManagerDice";
+export const oldestClientDiceId = "oldestClientDice";
+
+// Just a little helper, since we're going to request multiple objects.
+async function requestObjectStoreFromId<T>(request: RequestParser, runtime: IContainerRuntime, id: string) {
+    const fluidObjectRequest = RequestParser.create({
+        url: ``,
+        headers: request.headers,
+    });
+    return requestFluidObject<T>(
+        await runtime.getRootDataStore(id),
+        fluidObjectRequest);
+}
+
+const requestHandler: RuntimeRequestHandler =
+    async (request: RequestParser, runtime: IContainerRuntime) => {
+        const diceId = request.pathParts[0];
+        if (diceId === taskManagerDiceId || diceId === oldestClientDiceId) {
+            const dice = await requestObjectStoreFromId<IDiceRoller>(
+                request, runtime, diceId);
+            return { status: 200, mimeType: "fluid/object", value: dice };
+        }
+    };
+
+class TaskSelectionContainerRuntimeFactory extends BaseContainerRuntimeFactory {
+    constructor() {
+        super(registryEntries, [], [requestHandler]);
+    }
+
+    protected async containerInitializingFirstTime(runtime: IContainerRuntime) {
```

Comment:
This is exactly the code we want to avoid, having a singleton reference to an id set up by the container for retrieving this functionality. Can we demonstrate how at least for the oldestclientdiceroller version, we don't need to create this component singleton?

Reply:
I chatted with Leevi about this -- the important feature of this example is that the `DataObject`s (`OldestClientDiceRoller` and `TaskManagerDiceRoller`) are able to negotiate task selection locally to the `DataObject` without container involvement.  The means by which the `DataObject`s are instantiated (i.e. `createRootDataStore` in the container code vs. creation by a root `DataObject` as a subcomponent) is not core to the scenario, as task selection will work the same either way.